src/Compilers/Core/Portable/InternalUtilities/FatalError.cs

```diff
@@ -74,6 +75,13 @@ public static void OverwriteHandler(Action<Exception>? value)
             s_fatalHandler = value;
         }
 
+        private static bool IsCurrentOperationBeingCancelled(Exception exception, CancellationToken cancellationToken)
+#if NET20
+            => exception is OperationCanceledException cancellationException;
+#else
+            => exception is OperationCanceledException cancellationException && cancellationToken.IsCancellationRequested;
+#endif
```

Comment:
can we doc why this is necessary?  is IsCancellationRequested just not available?

Reply:
do we need to check if the OperationCanceledException is pointing at the specified cancellationToken.  If not, can we doc why that is right.  I would find it very weird/suspect if we got an OCE for a different CT.

Reply:
Correct, IsCancellationRequested is not defined in .NET 2.0. 

Re checking cancellation token - that'd be a follow-up once we actually agree on what to do. See https://github.com/dotnet/roslyn/pull/41477#discussion_r376189922 

I don't want to block this PR until that question is resolved.

Reply:
> I don't want to block this PR until that question is resolved.

That's fine, was just curious.